### ISSUE TITLE
better info around breathe parser error in init.py

### ISSUE THREAD

**Issue by indie** [michaeljones/breathe#366]
This error has come back on two completely different Linux systems (OSes), and 
it comes back the same on both. I think that is that it is a problem in the 
parser itself, not the code. My guess is that it is either the parser or the 
way I've documented a cpp code block in `.rst`, but it is not clear which. 

If it is the latter (my mistake), this needs better error messages to the user 
since `breathe.parser.ParserError: unclosed token: line 152, column 129` on an
 `__init.py__` file is kind of hard to debug. 

Here's the env 

```
Loaded extensions:
#   breathe (4.7.3) from /home/username/.local/lib/python3.5/site-packages/breathe/__init__.py
#   sphinx.ext.ifconfig (1.6.5) from /home/username/.local/lib/python3.5/site-packages/sphinx/ext/ifconfig.py
#   sphinx.ext.viewcode (1.6.5) from /home/username/.local/lib/python3.5/site-packages/sphinx/ext/viewcode.py
#   sphinx.ext.mathjax (1.6.5) from /home/username/.local/lib/python3.5/site-packages/sphinx/ext/mathjax.py
#   alabaster (0.7.10) from /home/username/.local/lib/python3.5/site-packages/alabaster/__init__.py
Traceback (most recent call last):
  File "/home/username/.local/lib/python3.5/site-packages/breathe/parser/__init__.py", line 48, in parse
    return self.cache[filename]
KeyError: '/home/username/libraries/project-name/doc/sphinx/source/../../doxygen/xml/index.xml'
```

And here is the error: 

```
.local/lib/python3.5/site-packages/breathe/parser/__init__.py", line 57, in parse
    raise ParserError(e, filename)
breathe.parser.ParserError: unclosed token: line 152, column 129
```

If the error is in the code for which I am writing documentation, it should not
say that breathe is the problem; that is what is says, though.  

The `.rst` file that it is trying to parse when it throws this error is of this 
format:

```rst
.. filename.rst

########
FileName
########

.. code-block:: cpp
   LaLa  // one line of cpp code short description


Description
===========

Some interesting long-description descriptive info written in .rst. Maybe one
paragraph.

Maybe many paragraphs.

Inputs
------

+-----------------+-------------------------+--------------------------------+
| Name            | Element Type            | Col 3                          |
+=================+=========================+================================+
| ``arg``         | Any                     | Any                            |
+-----------------+-------------------------+--------------------------------+

Outputs
-------

+-----------------+-------------------------+--------------------------------+
| Name            | Element Type            | Col 3                          |
+=================+=========================+================================+
| ``arg``         | Any                     | Any                            |
+-----------------+-------------------------+--------------------------------+

C++ Interface
=============

.. doxygenclass::foo::bar::Something
   :project: projectname
   :members: 



```

The more informative debug info in sphinx-log

```
File "/home/username/.local/lib/python3.5/site-packages/breathe/finder/core.py", line 87, in create_finder
    root = self.parser.parse(project_info)
  File "/home/username/.local/lib/python3.5/site-packages/breathe/parser/__init__.py", line 57, in parse
    raise ParserError(e, filename)
breathe.parser.ParserError: unclosed token: line 152, column 129
```

Thanks!

**Comment by indie** [comment]
P.S. The repo on which this is happening (so you can reproduce the slowness when the parser hits the `abs` op on the `build html`) is here:  https://github.com/NervanaSystems/ngraph/

**Comment by melvinvermeeren** [comment]
Now that I look again this issue appears to be very similar to #389, which has been fixed in v4.10.0.

Could you try updating to that version and see if the error message has been improved? It is likely this is caused by an old version of Doxygen, as noted in above issue. If it still exists could you try to post the relevant lines of the Doxygen XML if the error message makes this possible?

**Comment by melvinvermeeren** [comment]
Assuming fixed.
